projects/packages/assets/src/class-assets.php

```diff
@@ -223,4 +223,171 @@ public static function staticize_subdomain( $url ) {
+	 *  - `enqueue`:      (bool) Set true to enqueue the script immediately.
+	 *  - `async`:        (bool) Set true to register the script as async, like `Assets::enqueue_async_script()`
+	 *  - `in_footer`:    (bool) Set true to register script for the footer.
+	 *  - `media`:        (string) Media for the css file. Default 'all'.
+	 * @throws \InvalidArgumentException If arguments are invalid.
+	 */
+	public static function register_script( $handle, $path, $relative_to, array $options = array() ) {
+		if ( substr( $path, -3 ) !== '.js' ) {
+			throw new \InvalidArgumentException( '$path must end in ".js"' );
+		}
+
+		$dir      = dirname( $relative_to );
+		$base     = substr( $path, 0, -3 );
+		$options += array(
+			'asset_path'       => "$base.asset.php",
+			'css_path'         => "$base.css",
+			'css_dependencies' => array(),
+			'dependencies'     => array(),
+			'minify'           => false,
+			'enqueue'          => false,
+			'async'            => false,
+			'in_footer'        => false,
+			'media'            => 'all',
+		);
+
+		if ( $options['css_path'] && substr( $options['css_path'], -4 ) !== '.css' ) {
+			throw new \InvalidArgumentException( '$options[\'css_path\'] must end in ".css"' );
+		}
+
+		if ( isset( $options['nonmin_path'] ) ) {
+			$url = self::get_file_url_for_environment( $path, $options['nonmin_path'], $relative_to );
+		} else {
+			$url = plugins_url( $path, $relative_to );
+		}
+		$url = self::normalize_path( $url );
+		if ( null !== $options['minify'] ) {
+			$url = add_query_arg( 'minify', $options['minify'] ? 'true' : 'false', $url );
+		}
+
+		if ( $options['asset_path'] && file_exists( "$dir/{$options['asset_path']}" ) ) {
+			$asset                       = require "$dir/{$options['asset_path']}";
+			$options['dependencies']     = array_merge( $asset['dependencies'], $options['dependencies'] );
+			$options['css_dependencies'] = array_merge(
+				array_filter(
+					$asset['dependencies'],
+					function ( $d ) {
+						return wp_style_is( $d, 'registered' );
+					}
+				),
+				$options['css_dependencies']
+			);
+			$ver                         = isset( $options['version'] ) ? $options['version'] : $asset['version'];
+		} else {
+			$ver = isset( $options['version'] ) ? $options['version'] : filemtime( "$dir/$path" );
+		}
+
+		wp_register_script( $handle, $url, $options['dependencies'], $ver, $options['in_footer'] );
+		if ( $options['async'] ) {
+			self::instance()->add_async_script( $handle );
+		}
+
+		if ( $options['css_path'] && file_exists( "$dir/{$options['css_path']}" ) ) {
+			$csspath = $options['css_path'];
+			if ( is_rtl() ) {
+				$rtlcsspath = substr( $csspath, 0, -4 ) . '.rtl.css';
+				if ( file_exists( "$dir/$rtlcsspath" ) ) {
+					$csspath = $rtlcsspath;
+				}
+			}
+
+			$url = self::normalize_path( plugins_url( $csspath, $relative_to ) );
+			if ( null !== $options['minify'] ) {
+				$url = add_query_arg( 'minify', $options['minify'] ? 'true' : 'false', $url );
```

Comment:
Given that the `minify` string is only useful on WordPress.com right now (I think?), what would you think about only adding `minify=false` when on WordPress.com simple, via the `IS_WPCOM` constant?

Reply:
Personally I'd rather be consistent and always add it. It shouldn't hurt anything to include it.